packages/@react-stately/color/src/useColorAreaState.ts

```diff
@@ -84,34 +83,70 @@ export function useColorAreaState(props: ColorAreaProps): ColorAreaState {
   valueRef.current = color;
 
   let channels = useMemo(() => {
-    if (!xChannel) {
+    // determine the color space from the color value
+    let colorSpace = valueRef.current.getColorSpace();
+
+    if (colorSpace === 'rgb') {
+      if (!xChannel) {
+        switch (yChannel) {
+          case 'red':
+          case 'green':
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+            xChannel = 'blue';
+            break;
+          case 'blue':
+            xChannel = 'red';
+            break;
+          default:
+            xChannel = 'blue';
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+            yChannel = 'green';
+        }
+      } else if (!yChannel) {
+        switch (xChannel) {
+          case 'red':
+            yChannel = 'green';
+            break;
+          case 'blue':
+            yChannel = 'red';
+            break;
+          default:
+            xChannel = 'blue';
+            yChannel = 'green';
+        }
+      }
+    } else if (!xChannel) {
       switch (yChannel) {
-        case 'red':
-        case 'green':
-          xChannel = 'blue';
+        case 'hue':
+          xChannel = colorSpace === 'hsb' ? 'brightness' : 'lightness';
```

Comment:
are we 100% sure there will only be three color spaces in react spectrum? only hsb, rgb and hsl? If not I don't like this ternary. Never supporting YUV or CMYK?

Reply:
this is just for now